clients/iota-go/iotago/address.go

```diff
@@ -95,3 +95,9 @@ func (a *Address) UnmarshalJSON(data []byte) error {
 func (a Address) MarshalBCS() ([]byte, error) {
 	return a[:], nil
 }
+
+func (a Address) Clone() Address {
+	var ret [AddressLen]uint8
+	ret = a
+	return ret
+}
```

Comment:
```suggestion
func (a Address) Clone() (ret Address) {
	ret = a
	return
}
```

Also, aren't we passing `Address` as pointer everywhere?